GitCommands/Git/GitModule.cs

```diff
@@ -950,11 +950,15 @@ public GitRevision GetRevision(ObjectId? objectId = null, bool shortFormat = fal
             }
             else
             {
-                string message = ProcessDiffNotes(10);
+                string message = ProcessDiffNotes(startIndex: 10);
 
                 // commit message is not re-encoded by git when format is given
-                revision.Body = ReEncodeCommitMessage(message, revision.MessageEncoding);
-                revision.Subject = revision.Body?.Substring(0, revision.Body.IndexOfAny(new[] { '\r', '\n' })) ?? "";
+                // See also RevisionReader for parsing commit body
+                revision.Body = ReEncodeCommitMessage(message, revision.MessageEncoding)?.TrimStart().TrimEnd();
+                int endSubjectIndex = revision.Body?.IndexOf('\n') ?? -1;
+                revision.Subject = endSubjectIndex >= 0
+                    ? revision.Body.Substring(0, endSubjectIndex).TrimEnd()
+                    : revision.Body ?? "";
```

Comment:
We now have access to `ReadOnlySpan<char>`, so it would be significantly more performant to use it and splice as necessary. E.g. pseudo code:
```suggestion
                // See also RevisionReader for parsing commit body
                string body = ReEncodeCommitMessage(message, revision.MessageEncoding); // don't return null, return "" instead
                revision.Body = body;
                
                ReadOnlySpan<char> span = body.AsSpan();
                int endSubjectIndex = span.IndexOf('\n');
                revision.Subject = endSubjectIndex >= 0
                    ? span.Substring(0, endSubjectIndex).TrimEnd()
                    : "";
```

Reply:
This is not performance critical, but can change.
Adopting slightly (for instance not changing `ReEncodeCommitMessage()` - yet at least)